collector/lib/NetworkConnection.h

```diff
@@ -357,8 +357,14 @@ class ContainerEndpoint {
   const std::shared_ptr<Process> originator() const { return originator_; }
 
   bool operator==(const ContainerEndpoint& other) const {
-    return container_ == other.container_ && endpoint_ == other.endpoint_ && l4proto_ == other.l4proto_ &&
-           originator_ == other.originator_;
+    if (container_ != other.container_ || endpoint_ != other.endpoint_ || l4proto_ != other.l4proto_) return false;
+    if (originator_ == nullptr && other.originator_ != nullptr) return false;
+    if (originator_ != nullptr && other.originator_ == nullptr) return false;
+    if (originator_ == nullptr && other.originator_ == nullptr) return true;
```

Comment:
You can probably re-write this to be a bit more readable:
```suggestion

    if (originator_ == nullptr) {
      if (other.originator_ != nullptr) {
        return false;
      }
      return true;
    }
    
    if (other.originator_ == nullptr ) return false;

```
Note that I also added a few empty lines around the change so that it separates itself from the other checks. The end result takes up more lines, but is much easier to follow IMO.

Reply:
Actually, on second thought
```suggestion

    if (originator_ == nullptr) {
      return other.originator_ == nullptr
    }
    
    if (other.originator_ == nullptr ) return false;

```

Reply:
Done